integration_test/functions/src/index.ts

```diff
@@ -33,8 +34,37 @@ export const integrationTests: any = functions.https.onRequest((req: Request, re
       admin.auth().deleteUser(userRecord.uid);
     }),
   ]).then(() => {
-    // On test completion, redirect the user to the Firebase RTDB dashboard, to
-    // see the results stored there.
-    resp.redirect(`https://${process.env.GCLOUD_PROJECT}.firebaseio.com/testRuns/${testId}`);
+    // On test completion, check that all tests pass and reply "PASS", or provide further details.
+    console.log('Waiting for all tests to report they pass...');
+    let ref = admin.database().ref(`testRuns/${testId}`);
+    return new Promise((resolve, reject) => {
+      let testsExecuted = 0;
+      ref.on('child_added', (snapshot) => {
+        testsExecuted += 1;
+        if (!snapshot.val().passed) {
+          reject(new Error(`test ${snapshot.key} failed; see database for details.`));
+          return;
+        }
+        console.log(`${snapshot.key} passed (${testsExecuted} of ${numTests})`);
+        if (testsExecuted < numTests) {
+          // Not all tests have completed. Wait longer.
+          return;
+        }
+        // All tests have passed!
+        resolve();
+      });
+    }).then(() => {
+      ref.off();  // No more need to listen.
+      return Promise.resolve();
+    }).catch(err => {
+      ref.off();  // No more need to listen.
+      return Promise.reject(err);
+    });
+  }).then(() => {
+    console.log('All tests pass!');
+    resp.status(200).send('PASS');
+  }).catch(err => {
+    console.log(`Some tests failed: ${err}`);
+    resp.status(500).send(`FAIL - details at https://${process.env.GCLOUD_PROJECT}.firebaseio.com/testRuns/${testId}`);
```

Comment:
Nits:
1. the GCLOUD_PROJECT is not necessarily the subdomain to firebaseio.com; you can use functions.config().firebase.databaseUrl.
2. I think the URL has to end in .json if you're linking to a database URL.
3. You might actually want to consider a pantheon so it will try to load the database using the developer credentials rather than unauthenticated credentials.